test/jasmine/tests/hover_label_test.js

```diff
@@ -805,6 +805,40 @@ describe('hover info', function() {
                     name: 'trace one'
                 });
             })
+            .then(function() {
+                var nl = [[0, 0, 0], [0, 0, 0]];
+                return Plotly.restyle(gd, 'hoverlabel.namelength', [nl, nl]);
+            })
+            .then(function() {
+                // hoverlabel.namelength has no effect when <extra></extra> is set
```

Comment:
N.B. here `namelength` is set to 0, but `hovertemplate` is set with content inside `<extra>`. Here `namelength` is ignore and the extra (name) label still shows up.

Please let me know if you think `namelength` should (essentially) override `<extra>`.